File: Data Structures & Algorithms/majority-element/submission-1.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        nums_map = {}
        k = len(nums) / 2

        if len(nums) == 1:
            return nums[0]

        for num in nums:
            if num in nums_map:
                nums_map[num] += 1
                if nums_map[num] > k:
                    return num
            else:
                nums_map[num] = 1
        
        return -1

File: Data Structures & Algorithms/set-mismatch/submission-1.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    findErrorNums(nums: number[]): number[] {
        const map: Map<number, number> = new Map();
        let duplicate = -1;
        let missing = -1;

        for (const num of nums) {
            if (!map.has(num)) {
                map.set(num, 1);
            } else {
                map.set(num, map.get(num) + 1);
            }
        }

        for (let i = 1; i <= nums.length; i++) {
            if (map.has(i) && map.get(i) > 1) duplicate = i;
            if (!map.has(i)) missing = i;
        }


        return [duplicate, missing];
    }
}